reVX/least_cost_xmission/cost_creator.py

```diff
@@ -0,0 +1,410 @@
+                mults_arr[mask] = lu_mult
+
+            if 'slope' in r_conf:
+                r_slope = slope[mask]
+                slope_mult = self._compute_slope_mult(r_slope, r_conf['slope'])
+                mults_arr[mask] = mults_arr[mask] * slope_mult
+
+        # Calculate multipliers for regions not defined in `config`
+        logger.debug('Processing default region')
+        if d_mults is None:
+            d_mults = {}
+        default_mask = ~regions_mask
+
+        if 'land_use' in d_mults:
+            rlu = land_use[default_mask]
+            lu_mult = self._compute_land_use_mult(rlu, d_mults['land_use'],
+                                                  land_use_classes)
+            mults_arr[default_mask] = lu_mult
+
+        if 'slope' in d_mults:
+            r_slope = slope[default_mask]
+            slope_mult = self._compute_slope_mult(r_slope, d_mults['slope'])
+            mults_arr[default_mask] = mults_arr[default_mask] * slope_mult
+
+        # Set water multiplier last so we don't get super high multipliers at
+        # water body boundaries next to steep slopes
+        mults_arr[land_use == WATER_NLCD_CODE] = WATER_MULT
+
+        return mults_arr
+
+    def compute_base_line_costs(self, capacity, base_line_costs, cell_size):
+        """
+        Get base line cost per cell raster for a given voltage
+
+        Parameters
+        ----------
+        capacity : int
+            Desired line capacity (MW). This must be in base_line_costs.
+        base_line_costs : dict
+            Base line cost per mile for ISOs
+        cell_size : int
+            Raster cell size for all layers (meters). Assumed to be square.
+
+        Returns
+        -------
+        base_cost : numpy.ndarray
+            Cost per cell raster in same shape as iso_regions
+        """
+        base_cost = np.full(self._iso_regions.shape, float('inf'),
+                            dtype=np.float32)
+
+        for iso in base_line_costs:
+            logger.info(f'Processing costs for {iso} for {capacity}MW')
+            iso_code = self._iso_lookup[iso]
+            cost_per_mile = base_line_costs[iso][str(capacity)]
+            cost_per_cell = cost_per_mile / METERS_IN_MILE * cell_size
+            logger.debug(f'$/mile is {cost_per_mile}, $/cell is '
+                         f'{cost_per_cell}')
+            mask = self._iso_regions == iso_code
+            base_cost[mask] = cost_per_cell
+
+        return base_cost
+
+    def save_layer(self, layer_name, data):
+        """
+        Save data to h5 file
+
+        Parameters
+        ----------
+        layer_name : str
+            Name of layer to save in h5
+        data : np.ndarray
+            Data to savein h5
+        """
+        logger.debug(f'Saving {layer_name} to h5')
+        self._profile['nodata'] = None
+        self._write_layer(self._excl_h5, layer_name, self._profile, data)
+
+    def create_geotiff(self, tiff_dir, geotiff_fpath, data):
```

Comment:
Simplify this to be just:
```
def create_geotiff(self, geotiff_fpath, data):
```
There is no need to do an os.path.join inside this method.

Reply:
Keeping os.path.join inside create_geotiff is much DRYer and keeps the code in build_cost_rasters() shorter. I'd prefer to save the tiff_dir as an attribute to further simply the function arguments. Thoughts?

Reply:
I don't fully follow. Since this is a public method it seem much more straight forward to just request the full geotiff path and data to write (for anyone using this class that is not us). Then in the `build_cost_rasters` class method the tiff_dir is an input value so its a local variable, and we can create the custom file paths with `os.path.join` inside the `if save_geotfif` logic on lines 135 and 142.